src/Examine.AzureDirectory/ReadOnlyAzureDirectoryFactory.cs

```diff
@@ -1,14 +1,58 @@
+using System;
+using System.Configuration;
+using System.IO;
+using System.Web;
+using Examine.LuceneEngine.DeletePolicies;
 using Examine.LuceneEngine.Directories;
+using Examine.LuceneEngine.MergePolicies;
+using Examine.LuceneEngine.MergeShedulers;
+using Microsoft.Azure.Storage;
 
 namespace Examine.AzureDirectory
 {
     /// <summary>
     /// The <see cref="IDirectoryFactory"/> for storing master index data in Blob storage for user on the server that only reads from the index
     /// </summary>
-    public class ReadOnlyAzureDirectoryFactory : AzureDirectoryFactory
+    public class ReadOnlyAzureDirectoryFactory : AzureDirectoryFactory, IDirectoryFactory
     {
-        public ReadOnlyAzureDirectoryFactory() : base(isReadOnly:true)
-        {   
+        private readonly bool _isReadOnly = true;
+        public override Lucene.Net.Store.Directory CreateDirectory(DirectoryInfo luceneIndexFolder)
+        {
+            var indexFolder = luceneIndexFolder;
```

Comment:
All of this code seems just duplicated code, why do we need AzureReadOnlyDirectory instead of just using the isReadOnly flag?

Reply:
@Shazwazza as you can check In time of sync I am creating second index which is sync and when it is done I swapping them, it is fixing issue when you try to read and override index in same time, and adding that logic to main AzureDirectoryFactory imho would be bloating class there.

Reply:
true, but in same I am fun of more explicit methods as using flag is not good idea in my idea when we call different directory underhood, which in future could have additional settings etc.